Shopist/Shopist/API.swift

```diff
@@ -37,6 +82,28 @@ final class API {
         get(urlString, completion: completion)
     }
 
+    func checkout(with discountCode: String?, payment: Payment = Payment(), completion: @escaping Completion<Payment.Response>) {
+        var url = "\(Self.apiURL)/checkout.json"
+        if let someCode = discountCode {
+            url.append("?coupon_code=\(someCode)")
+        }
+        var request = try! URLRequest(url: url, method: .post)
+        request.httpBody = try! jsonEncoder.encode(payment)
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        httpClient.request(request).response { response in
+            if let someError = response.error {
+                completion(.failure(someError))
+            } else if let someData = response.data {
+                do {
+                    let paymentResponse = try self.jsonDecoder.decode(Payment.Response.self, from: someData)
+                    completion(.success(paymentResponse))
+                } catch {
+                    completion(.failure(error))
```

Comment:
This leads to decoding error presented in UI when server responds with `500` and HTML page. How about considering the response code and improving the error message?

Reply:
resolved ✅ 

this line doesn't look outdated but adding `validate()` to line 93 creates a response error in case of non-2XX HTTP status code
and the error message is smth like `HTTP Status Code 500 is not an acceptable status code`

Reply:
Awesome 🚀